docs/usage/self-hosted-configuration.md

```diff
@@ -324,31 +324,92 @@ This private key is used to decrypt config files.
 The corresponding public key can be used to create encrypted values for config files.
 If you want a simple UI to encrypt values you can put the public key in a HTML page similar to <https://renovatebot.com/encrypt>.
 
-To create the key pair with OpenSSL use the following commands:
+To create the key pair with GPG use the following commands:
 
-- `openssl genrsa -out rsa_priv.pem 4096` for generating the private key
-- `openssl rsa -pubout -in rsa_priv.pem -out rsa_pub.pem` for extracting the public key
+- `gpg --full-generate-key` and follow the prompts to generate a key. Name and email are not important to Renovate, and do not configure a passphrase. Use a 4096bit key.
 
-To encrypt a secret with OpenSSL use the following command:
+<details><summary>key generation log</summary>
 
-```bash
-echo 'actual-secret' | openssl rsautl -encrypt -pubin -inkey rsa_pub.pem | base64
 ```
+❯ gpg --full-generate-key
+gpg (GnuPG) 2.2.24; Copyright (C) 2020 Free Software Foundation, Inc.
+This is free software: you are free to change and redistribute it.
+There is NO WARRANTY, to the extent permitted by law.
+
+Please select what kind of key you want:
+   (1) RSA and RSA (default)
+   (2) DSA and Elgamal
+   (3) DSA (sign only)
+   (4) RSA (sign only)
+  (14) Existing key from card
+Your selection? 1
+RSA keys may be between 1024 and 4096 bits long.
+What keysize do you want? (3072) 4096
+Requested keysize is 4096 bits
+Please specify how long the key should be valid.
+         0 = key does not expire
+      <n>  = key expires in n days
+      <n>w = key expires in n weeks
+      <n>m = key expires in n months
+      <n>y = key expires in n years
+Key is valid for? (0)
```

Comment:
Is it a smart idea to make a Renovate bot key that doesn't expire? I thought usually its a good idea to rotate keys like this? But maybe it's really inconvenient if the bot suddenly stops working because the key expired on a set schedule?

You probably know better than me what is a good idea and what isn't, so I'll let you decide. 😉

Reply:
It's ok if they want the key to expire. Ref: https://community.progress.com/s/article/Can-a-PGP-encryption-key-be-used-after-the-expiration-date